### DISCUSSION TITLE
How to use Stats API in self-hosted?

### DISCUSSION THREAD

**Discussion by erdempython** [plausible/analytics#3618] [Self-Hosted Support] [answered]
I use Stats API in cloud-hosted using endpoints like below:

curl 'https://plausible.io/api/v1/stats/realtime/visitors?site_id=$SITE_ID'
  -H "Authorization: Bearer ${TOKEN}"

Here, site_id is my domain name (i.e. domainName.com).

I built my integration based on this. But my customers are asking for support for self-hosted sites. So when the site is self-hosted, how to change the URL? Will it be like this? I have no way of testing, any help is greatly appreciated.

curl 'https://domainName.com/plausible/api/v1/stats/realtime/visitors?site_id=$SITE_ID'
  -H "Authorization: Bearer ${TOKEN}"

**Comment by ruslandoga** [comment, marked as answer]
👋 @erdempython 

It depends on the self-hosted setup, but if there is no reverse-proxy that rewrites paths, the URL would be

`https://domainname.com/api/v1/stats/realtime/visitors?site_id=$SITE_ID`

That is, the path would stay the same.

**Reply by erdempython** [reply]
Thanks a ton for the quick response, @ruslandoga. Do you think there is any opportunity for me to find a test account? I just need to run the query to see if this works before the release.

**Reply by ruslandoga** [reply]
A test account on a self-hosted instance? I have a small dataset on who.copycat.fun, I''ll send you a token if I figure out how to get one :)

**Reply by erdempython** [reply]
Wouww Ruslan, you are amazing! You can obtain an API key for your account by going to your user settings page [plausible.io/settings](https://plausible.io/settings).

**Reply by ruslandoga** [reply]
QN04hpgQFCrqWnmXZeOnxnIy5Tsqw5W9gsfteVTe7woP8D3v6gBqWmf-j8hxB1tq

<img width="767" alt="Screenshot 2023-12-12 at 21 20 03" src="https://github.com/plausible/analytics/assets/67764432/d4f8942e-8317-4798-a143-47019038b6db">

**Reply by ruslandoga** [reply]
Sites:

- https://who.edify.space/stats.copycat.fun
- https://who.edify.space/dummy.site?period=all

**Reply by erdempython** [reply]
Hmm, I get a response in the form of a web page. But I need it in JSON format for my application. Let me know if there is a way to get a JSON response out of these. If not, thanks a lot for your effort. You have been really helpful.

**Reply by ruslandoga** [reply]
```console
$ curl -H 'authorization: Bearer QN04hpgQFCrqWnmXZeOnxnIy5Tsqw5W9gsfteVTe7woP8D3v6gBqWmf-j8hxB1tq' 'https://who.edify.space/api/v1/stats/realtime/visitors?site_id=stats.copycat.fun'
0
```

**Reply by erdempython** [reply]
Oh yea, this works. Ruslan, you are the man! Awesome help, thank you so much!

**Reply by erdempython** [reply]
I just needed to change site_id and it worked for those who need the trick.
